tsconfig.json

```diff
@@ -3,20 +3,18 @@
     "outDir": "./dist/",
     "sourceMap": true,
     "noImplicitAny": true,
-    "module": "commonjs",
-    "target": "es5",
+    "module": "esnext",
+    "moduleResolution": "node",
+    "target": "es2018",
     "allowJs": true,
     "esModuleInterop": true,
     "resolveJsonModule": true,
     "experimentalDecorators": true,
-    "strictNullChecks": true,
-    "types": ["jest"],
+    "importHelpers": true,
+    "strict": true,
+    "types": [],
     "jsx": "react",
-    "lib": [
-      "dom",
-      "es5",
-      "es2015"
-    ]
+    "lib": ["dom", "dom.iterable", "es2020"]
```

Comment:
This seems a little risky to me. If I understand it right, because TSC doesn't do any polyfilling itself. We could use String.matchAll in the code and it would compile fine and work fine on any es2020 browser. But if we tried to run it on an es2018 browser it would fail.

I don't know the right solution though. We could set target to es2020 so it would be more obvious that is the target with the comment that you can add core-js to support es2018.  Or we could just always add core-js. Or we could set the lib property to es2018.

Reply:
>I don't know the right solution though. We could set target to es2020 so it would be more obvious that is the target with the comment that you can add core-js to support es2018. Or we could just always add core-js. Or we could set the lib property to es2018.

My thinking was that it's reasonable to assume that for projects starting today that the overwhelming majority of our users can be assumed to be running browsers that are no older than three years old today, i.e. at least four years old based on your assumption of going live to users in a year. If we encounter the need to support older browsers (presumably because it crashed on such an older browser), we could address the polyfill question at that time. The polyfill question is a little bit tricky because `ts-polyfill` seems not to be maintained any more, `core-js` is quite heavy, and the approaches to using `core-js` efficiently often involve babel which we're not using. All of that said, I'm happy to drop this back to `es2018` rather than holding things up any longer.

Reply:
I'm sad that we can't just use es2020.  But because of the polyfill issue it does seem the safest to just use es2018.

I did see we could use pollyfill.io to automatically add anything the current browser doesn't have. So that would be a way to safely use es2020, but I don't really trust polyfill.io since it is a service that doesn't seem to have any direct revenue stream.